aldryn_forms/storage_backends.py

```diff
@@ -0,0 +1,93 @@
+import abc
+
+import six
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.utils.module_loading import import_string
+from django.utils.translation import ugettext_lazy as _
+
+
+DEFAULT_ALDRYN_FORMS_STORAGE_BACKENDS = {
+    'default': 'aldryn_forms.storage_backends.DefaultStorageBackend',
+    'no_storage': 'aldryn_forms.storage_backends.NoStorageBackend',
+}
+DEFAULT_ALDRYN_FORMS_DEFAULT_STORAGE_BACKEND = 'default'
+ALDRYN_FORMS_STORAGE_BACKEND_KEY_MAX_SIZE = 15
+
+
+def get_storage_backends():
+    if hasattr(settings, 'ALDRYN_FORMS_STORAGE_BACKENDS'):
+        backends = settings.ALDRYN_FORMS_STORAGE_BACKENDS
+    else:
+        backends = DEFAULT_ALDRYN_FORMS_STORAGE_BACKENDS
+
+    try:
+        backends = {k: import_string(v) for k, v in backends.items()}
+    except ImportError as e:
+        raise ImproperlyConfigured(_('Invalid settings.ALDRYN_FORMS_STORAGE_BACKENDS. {exc}').format(exc=e))
+
+    if any(filter(lambda x: len(x) > ALDRYN_FORMS_STORAGE_BACKEND_KEY_MAX_SIZE, backends.keys())):
+        raise ImproperlyConfigured(_(
+            'Invalid settings.ALDRYN_FORMS_STORAGE_BACKENDS. Ensure all keys are no longer than {qty} characters.'
+        ).format(qty=ALDRYN_FORMS_STORAGE_BACKEND_KEY_MAX_SIZE))
+
+    if any(filter(lambda x: not(issubclass(x, BaseStorageBackend)), backends.values())):
+        raise ImproperlyConfigured(_(
+            'Invalid settings.ALDRYN_FORMS_STORAGE_BACKENDS. '
+            'All classes must derive from aldryn_forms.storage_backends.BaseStorageBackend'
+        ))
+
+    try:
+        [x() for x in backends.values()]  # check abstract base classes sanity
+    except TypeError as e:
+        raise ImproperlyConfigured('{}'.format(e))
+
+    return backends
+
+
+def storage_backend_choices(*args, **kwargs):
+    choices = tuple([(key, klass.verbose_name) for key, klass in get_storage_backends().items()])
+    return sorted(choices, key=lambda x: x[1])
+
+
+def storage_backend_default(*args, **kwargs):
+    default_backend = getattr(
+        settings, 'ALDRYN_FORMS_DEFAULT_STORAGE_BACKEND', DEFAULT_ALDRYN_FORMS_DEFAULT_STORAGE_BACKEND
```

Comment:
I don't think this key should be configurable.
considering you always have to override the dictionary.
best to keep it always as default for consistency.